src/middlewared/middlewared/plugins/auth.py

```diff
@@ -1,74 +1,238 @@
 import crypt
+from datetime import datetime, timedelta
+import random
+import re
 import socket
+import string
 import subprocess
 import time
-import uuid
 
 from middlewared.schema import Dict, Int, Str, accepts
-from middlewared.service import Service, no_auth_required, pass_app, private
+from middlewared.service import Service, filterable, filter_list, no_auth_required, pass_app, private
 from middlewared.utils import Popen
 
 
-class AuthTokens(object):
-
+class TokenManager:
     def __init__(self):
-        # Keep two indexes, one by token id and one by session id
-        self.__tokens = {}
-        self.__sessionid_map = {}
+        self.tokens = {}
 
-    def get_token(self, token_id):
-        # Get token entry from token id
-        return self.__tokens.get(token_id)
+    def create(self, ttl, attributes=None):
+        token = "".join(random.choice(string.ascii_letters + string.digits) for _ in range(64))
```

Comment:
This also changed away from uuid. What is the reason for changing that?

Reply:
Because UUIDs look terrible as access tokens, like «we don't know any other way to generate a random string»

Reply:
Well, thats a personal opinion, both are just as random.

Reply:
What is the technical reasoning for the change?